### PR TITLE
Add more USB HID Class constants

### PR DESCRIPTION
**Pull request by fornellas**
Add more USB HID Class constants based on [Device Class Definition for Human Interface Devices (HID)  v1.11](https://usb.org/sites/default/files/documents/hid1_11.pdf).

### CODE REVIEW

**Comment by fornellas** [comment]
Ping. Is this good to merge or is there a feeling that it requires changes? Thanks.

File: include/libopencm3/usb/hid.h [hunk 1]
```diff
@@ -42,8 +42,30 @@ LGPL License Terms @ref lgpl_license
...
+#define USB_HID_SUBCLASS_BOOT_INTERFACE 1
+
+#define USB_HID_PROTOCOL_NONE 0
+#define USB_HID_PROTOCOL_KEYBOARD 1
+#define USB_HID_PROTOCOL_MOUSE 2
+
+#define USB_HID_DT_HID 0x21
+#define USB_HID_DT_REPORT 0x22
+#define USB_HID_DT_PHYSICAL 0x23
+// old names without USB_HID_ prefix
```
Comment (karlp) [inline include/libopencm3/usb/hid.h]: no // comments.  Can we get attributes on these to make them pop warnings?

File: include/libopencm3/usb/hid.h [hunk 2]
```diff
@@ -42,8 +42,30 @@ LGPL License Terms @ref lgpl_license
...
+#define USB_DT_HID USB_HID_DT_HID
+#define USB_DT_REPORT USB_HID_DT_REPORT
```
Comment (karlp) [inline include/libopencm3/usb/hid.h]: no // comments please, and if possible, see if you can get an attribute on the old name that triggers a warning.

**Comment by karlp** [comment]
Otherwise looks good

File: include/libopencm3/usb/hid.h (hunk 2 above)
Comment (fornellas) [reply include/libopencm3/usb/hid.h]: I'm not sure how to expand a macro for an integer into `#warning "whatever"`, if that's what you meant, open to suggestions.

Personally I'd just remove these 2 lines, as the errors that will surface should be pretty obvious to amend. I can do the PR for libopencm3-examples with the necessary adjustments as well.

**Comment by karlp** [comment]
merged in f2f50837 thanks!

**Pull request closed without merging**